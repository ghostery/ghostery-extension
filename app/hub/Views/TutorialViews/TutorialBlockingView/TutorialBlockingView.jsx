/**
 * Tutorial Blocking View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 *
 * ToDo: Update this file.
 */

import React from 'react';

/**
 * @class Implement the Tutorial Blocking View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
const TutorialBlockingView = () => <div>Ghostery Hub - Tutorial Blocking View</div>;

export default TutorialBlockingView;
