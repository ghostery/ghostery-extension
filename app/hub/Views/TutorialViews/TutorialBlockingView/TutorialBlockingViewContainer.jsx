/**
 * Tutorial Blocking View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TutorialBlockingView from './TutorialBlockingView';

/**
 * @class Implement the Tutorial Blocking View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class TutorialBlockingViewContainer extends Component {
	constructor(props) {
		super(props);

		const title = t('hub_tutorial_page_title_blocking');
		window.document.title = title;

		const { index } = props;
		props.actions.setTutorialNavigation({
			activeIndex: index,
			hrefPrev: `/tutorial/${index - 1}`,
			hrefNext: `/tutorial/${index + 1}`,
			hrefDone: '/home',
			textPrev: t('previous'),
			textNext: t('next'),
			textDone: t('hub_tutorial_exit_flow'),
		});
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Tutorial Blocking View of the Hub app
	 */
	render() {
		return <TutorialBlockingView />;
	}
}

// PropTypes ensure we pass required props of the correct type
TutorialBlockingViewContainer.propTypes = {
	index: PropTypes.number.isRequired,
	actions: PropTypes.shape({
		setTutorialNavigation: PropTypes.func.isRequired,
	}).isRequired,
};

export default TutorialBlockingViewContainer;
