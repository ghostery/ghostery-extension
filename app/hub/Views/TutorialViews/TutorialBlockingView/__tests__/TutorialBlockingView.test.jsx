/**
 * Tutorial Blocking View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TutorialBlockingView from '../TutorialBlockingView';

describe('app/hub/Views/TutorialViews/TutorialBlockingView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('tutorial blocking view is rendered correctly', () => {
			const component = renderer.create(<TutorialBlockingView />).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const component = shallow(<TutorialBlockingView />);
			expect(component.find('.TutorialBlockingView').length).toBe(1);
			expect(component.find('.TutorialBlockingView__image').length).toBe(2);
			expect(component.find('.TutorialView__keyText').length).toBe(3);
		});
	});
});
