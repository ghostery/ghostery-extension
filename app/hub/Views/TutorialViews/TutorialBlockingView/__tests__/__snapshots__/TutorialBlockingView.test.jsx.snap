// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/TutorialViews/TutorialBlockingView component Snapshot tests with react-test-renderer tutorial blocking view is rendered correctly 1`] = `
<div
  className="TutorialBlockingView row align-center-middle flex-container"
>
  <div
    className="columns small-11 small-offset-1 medium-6"
  >
    <div
      className="TutorialView__imageTitle"
    >
      detailed_view
    </div>
    <img
      alt="detailed_view"
      className="TutorialBlockingView__image blocking-detailed"
      src="/app/images/hub/tutorial/blocking-detailed.png"
    />
    <div>
      <div
        className="TutorialView__imageTitle"
      >
        hub_tutorial_detailed_expanded_view
      </div>
      <img
        alt="hub_tutorial_detailed_expanded_view"
        className="TutorialBlockingView__image blocking-detailed-expanded"
        src="/app/images/hub/tutorial/blocking-detailed-expanded.png"
      />
    </div>
  </div>
  <div
    className="columns small-12 medium-5 large-4"
  >
    <div
      className="TutorialView__title"
    >
      hub_tutorial_blocking_title
    </div>
    <div
      className="TutorialView__tagline"
    >
      hub_tutorial_blocking_tagline
    </div>
    <div
      className="TutorialBlockingView__key flex-container flex-dir-column"
    >
      <div
        className="TutorialView__keyText allow"
      >
        hub_tutorial_blocking_allow
      </div>
      <div
        className="TutorialView__keyText block"
      >
        hub_tutorial_blocking_block
      </div>
      <div
        className="TutorialView__keyText block-global"
      >
        hub_tutorial_blocking_block_always
      </div>
    </div>
  </div>
</div>
`;
